python_tests/test_python_frontend.py

```diff
@@ -2359,6 +2359,49 @@ def fusion_func(fd: FusionDefinition) -> None:
         # Just test that this executes, not that it's correct
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
 
+    # Test that deterministic random ops (uniform, normal) give same results as
+    # their stochastic versions
+    def test_deterministic_random(self):
+        input_size = [5, 9]
+        dtype = torch.float32
+        device = "cuda"
+        inputs = [
+            torch.randn(*input_size, device=device, dtype=dtype),
+        ]
+        param1 = 0.3  # low or mean
+        param2 = 1.7  # hi or sigma
+
+        for randopname in ["uniform", "normal"]:
+
+            def fusion_func(fd: FusionDefinition):
+                # Note: this is awkward, but we need to set the random seed
+                # each time the fusion is executed otherwise subsequent calls
+                # will compute t1 after advancing the seed
+                torch.manual_seed(0)
```

Comment:
This also might affect all other tests. We can use a special context manager to localize this seed manipulation:
```py
with torch.random.fork_rng(devices=[device.index]):
    torch.manual_seed(0)
    ...
```